stat.go

```diff
@@ -62,7 +62,7 @@ type Stat struct {
 	// Summed up cpu statistics.
 	CPUTotal CPUStat
 	// Per-CPU statistics.
-	CPU []CPUStat
+	CPU map[int64]*CPUStat
```

Comment:
I think it would be better to just use a `CPUStat` instead of pointer `*CPUStat` here.  The struct is relatively small, so I think memory shouldn't be an issue.

Reply:
We started with `CPUStat` initially, the problem is in node_exporter, it tracks `CPUStat` with the same data structure. Working with references makes both implementation clean. If pointer should be avoided here, I could see what I can do in the node_exporter side.

Reply:
@alexjx Why does it make implementation cleaner?

Reply:
Sorry for the long delay. What I thought back then was that by making it a pointer, we have the ablity to test if the value "exist" in certain senarios. But reviewing node_exporter implementation, It seem the case is rare. I will try updating the patch.